catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/SourcePoller.java

```diff
@@ -403,21 +227,161 @@ public void setCatalogStores(final List<CatalogStore> catalogStores) {
   }
 
   /**
-   * @param pollIntervalMinutes the interval (in minutes) at which to recheck the availability of
-   *     the {@link Source}s
-   * @throws IllegalArgumentException if {@code pollIntervalMinutes} is less than {@code 1} or is
-   *     greater than {@link Integer#MAX_VALUE}
+   * This method is used to keep the cache up to date. The current {@link Source}s configured by
+   * {@link #setConnectedSources(List)}, {@link #setFederatedSources(List)}, {@link
+   * #setCatalogProviders(List)}, and {@link #setCatalogStores(List)} are discovered. This method
+   * blocks until the availability of each {@link Source} is checked and the cache is updated.
    */
-  public void setPollIntervalMinutes(final long pollIntervalMinutes) {
-    final int minimumPollIntervalMinutes = 1;
-    if (pollIntervalMinutes < minimumPollIntervalMinutes) {
+  private synchronized void pollAllSources() {
+    try {
+      final Map<SourceKey, Source> currentSourceKeysAndSourcesMap =
+          getCurrentSourceKeysAndSourcesMap();
+
+      removeSourceKeysFromCache(currentSourceKeysAndSourcesMap);
+      addSourceKeysToCache(currentSourceKeysAndSourcesMap);
```

Comment:
✏️ I think we had a little more information conveyed in the method names we had before. Something like removeExtraneousSourceKeysFromCache() or removeStaleSourceKeysFromCache() and addMissingSourceKeysToCache() or addNewSourceKeysToCache() might be more descriptive.